### ISSUE TITLE
ImageColumn needs a short circuit to not call apis for temporary urls

### ISSUE THREAD

**Issue by blite** [filamentphp/filament#4304]
### Package

filament/tables

### Package Version

v2.16.19

### Laravel Version

v9.33.0

### Livewire Version

v2.10.7

### PHP Version

PHP 8.1.8

### Problem description

While using S3 all image columns perform a s3 call to check the ACL of the object before deciding whether to show the object with a temporary url or not.  

This results in s3 ImageColumn tables loading very slowly.  

AWS is currently trying to get users to not even use these ACL's and instead enforce private / public on a bucket wide or policy basis and not on an object basis.   $storage->getVisibility ignores policy based ACL's and only looks for object based ones.

Since there are no comments at https://github.com/filamentphp/filament/blob/2.x/packages/tables/src/Columns/ImageColumn.php#L122

We can't really divine why it is checking the column's settings then the object.  It would be logical to simply do:

```
if ($this->getVisibility() != 'public' && $storage->getVisibility($state) === 'private') {
```
It seems likely to me that the original || should have been &&
but since we don't know why that === private is there we would be safest to change it to:

```
if ($this->getVisibility() != 'public'  && ($this->getVisibility() === 'private' || $storage->getVisibility($state) === 'private')) {
```

### Expected behavior

s3 images should not not have to do an acl check per row if the images are public

### Steps to reproduce

setup s3
setup imagecolumn with s3
notice the slow

### Reproduction repository

https://github.com/filamentphp/filament/

**Comment by blite** [comment]
@danharrin there is no way to make a reproduction repository for this as you can't share s3 credentials in a repo

**Comment by danharrin** [comment]
tbh we can probably get rid of the `getVisibility()` check altogether in this case, that is safest. submit a pr? also for the `BaseFileUpload` probably

**Comment by blite** [comment]
well it was specifically added in this commit https://github.com/filamentphp/filament/commit/302ba6d7e11c4543aded6063907700f11538cebf#diff-6fd324eee2d85802ecea21993004627117816f6b06116d48d68ce0b74a2ea0f9 

notice in spatie column it has && which makes sense to me and makes me think the || is a typo

**Comment by blite** [comment]
It seems that having acl's disabled breaks the fileupload.  S3 pushes you into disabling acl's currently but that could change at any time.  It simply appears to have uploaded the file and never moves it from the temporary upload folder (likely an upstream problem, but maybe one you should be aware of for documentation).  

<img width="843" alt="Screen Shot 2022-09-30 at 3 53 27 PM" src="https://user-images.githubusercontent.com/442226/193366833-5f9ae8d4-54e6-4b99-bb2b-7b651ea88c28.png">

I personally think the getVisibility check needs the && and should be added to BaseFileUpload as well.  I feel like this will definitely break it for someone though.

It seems like to me that temporaryUrls should be opt in and someone who actually cares about them is likely going to be more diligent about things.

**Comment by blite** [comment]
So found the likely reason this was maybe configured as private always:


with S3 there seems to be no vary: origin header if you didn't send an origin header originally.  https://stackoverflow.com/questions/31732533/s3-cors-always-send-vary-origin

So if you don't first retrieve the image with cors ie using ImageColumn, then retrieve it in FileUpload using cors, the second time it won't load with an error like "No 'Access-Control-Allow-Origin' header is present on the requested resource" in browsers error console.  

Solution is to load all ImageColumn with ```->extraImgAttributes(['crossorigin' => 'anonymous'])```

I don't think any changes need to be made for this, but it was an annoying bug and commenting here incase anyone else runs into it.

**Comment by danharrin** [comment]
But it still works out of the box if you use `visibility('private')`, right? Because S3 still allows generating temporary URLs for public images

**Comment by blite** [comment]
I use public which is the default, private fixes this, and likely this explains why private was basically permanently on internally. 

The tldr is that regular public urls cache a non cors version which then breaks the cors loaded image in filepond.  

Overall with this extra attribute it is better performance because of no extra s3 calls. The bug is very hard to debug because having devtools open never allows it to present itself, and it depends on which version of the image you cached first (cors vs non cors). Apparently there are other workarounds with cloudfront etc.  A workaround on our end could be to add a cachebusting parameter in the filepond url, but that seems dirty. Also if I had just used a smaller thumbnail like I really should be doing, this also wouldn't be an issue.

**Comment by danharrin** [comment]
Do you think the attribute should just be present on all images by default?

**Comment by blite** [comment]
maybe, but it might need to handle the case for changing crossorigin to "use-credentials" https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin.  Also, it could cause bugs with images where you don't have a cors policy on the host they are served from (say you aren't using fileupload control at all).
